oauth_client.go

```diff
@@ -125,3 +126,40 @@ func (s *oAuthClients) Create(ctx context.Context, organization string, options
 
 	return oc, nil
 }
+
+type OAuthCLientDestroyOptions struct {
+	CLIENT_ID    *string `jsonapi:"primary,oauth-client"`
+	Organization *string `jsonapi:"primary,organization"`
+}
+
+func (o OAuthCLientDestroyOptions) valid() error {
+	if !validString(o.CLIENT_ID) {
+		return errors.New("CLIENT_ID required")
+	}
+	return nil
+}
+
+func (s *oAuthClients) Delete(ctx context.Context, options OAuthCLientDestroyOptions) (*OAuthClient, error) {
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("oauth-clients/%s", url.QueryEscape(*options.CLIENT_ID))
+
+	req, err := s.client.newRequest("DELETE", u, s)
+	if err != nil {
+		return nil, err
+	}
+
+	oc := &OAuthClient{}
```

Comment:
You are creating a new empty value and then returning that directly to the caller without unmarshalling any response data. And since this API call does not return any data, we should just omit this.